### PR TITLE
update go test version matrix

### PR DESCRIPTION
**Pull request by andyborn**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go-version: [1.17.x, 1.18.x]
+        go-version: [1.20.x, 1.21.x]
```
Comment (andyborn) [inline .github/workflows/test.yml:14]: should I bump the lint version too?  v1.20?

Comment (dim) [reply .github/workflows/test.yml:14]: latest pls

Comment (dim) [reply .github/workflows/test.yml:14]: I think you can just use `1.x`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
```
Comment (dim) [inline .github/workflows/test.yml:17]: upgrade to v4 pls, you can remove `cache: true` below as it's now the default

**Review by dim — CHANGES_REQUESTED** [review]

**Review by dim — APPROVED** [review]

**Pull request merged**